scrypto-unit/src/test_runner.rs

```diff
@@ -438,6 +441,71 @@ impl<'s, S: ReadableSubstateStore + WriteableSubstateStore> TestRunner<'s, S> {
         let receipt = self.execute_manifest(manifest, vec![signer_public_key]);
         receipt.new_component_addresses[0]
     }
+
+    pub fn set_current_epoch(&mut self, epoch: u64) {
+        self.system_kernel_call(|kernel| {
+            kernel
+                .invoke_method(
+                    Receiver::Ref(RENodeId::System),
+                    FnIdentifier::Native(NativeFnIdentifier::System(SystemFnIdentifier::SetEpoch)),
+                    ScryptoValue::from_typed(&SystemSetEpochInput { epoch }),
+                )
+                .unwrap()
+        });
+    }
+
+    pub fn get_current_epoch(&mut self) -> u64 {
+        let current_epoch: ScryptoValue = self.system_kernel_call(|kernel| {
+            kernel
+                .invoke_method(
+                    Receiver::Ref(RENodeId::System),
+                    FnIdentifier::Native(NativeFnIdentifier::System(SystemFnIdentifier::SetEpoch)),
+                    ScryptoValue::from_typed(&SystemGetCurrentEpochInput {}),
+                )
+                .unwrap()
+        });
+        scrypto_decode(&current_epoch.raw).unwrap()
+    }
+
+    /// Performs a kernel call through a kernel with `is_system = true`.
+    fn system_kernel_call<F>(&mut self, fun: F) -> ScryptoValue
+    where
+        F: FnOnce(
+            &mut Kernel<DefaultWasmEngine, DefaultWasmInstance, SystemLoanFeeReserve>,
+        ) -> ScryptoValue,
+    {
+        let tx_hash = hash(self.next_transaction_nonce.to_string());
+        let mut substate_store = self.execution_stores.get_output_store(0);
+        let mut wasm_engine = DefaultWasmEngine::new();
+        let mut wasm_instrumenter = WasmInstrumenter::new();
```

Comment:
Minor: Wasm Engine and Wasm Instrumenter should exist in `self` already

Reply:
Yes you are right, changing that :)